subsystems/DrivetrainBase.java

```diff
@@ -1,58 +1,133 @@
+        /* PID SETUP */
+        super(new PIDController(constants.turnFPID.kP, constants.turnFPID.kI, constants.turnFPID.kD));        
+        // disable PID control on start
+        this.disable();
+
+        /* VARIABLE DECLARATION */
+        // set all variables declared at the top to those given in the constructor (mostly constants)
+        this.constants = constants;
+        Rotation2d currentAngle = Rotation2d.fromDegrees(getHeading360());
+        this.odometry = new DifferentialDriveOdometry(currentAngle);
+        this.poseHistory =  new LimitedPoseMap(constants.historyLimit);
+        this.turnFPID = constants.turnFPID;
+        this.turnFPIDThrottle = constants.turnPIDThrottle;
+        this.invertGyro = constants.invertGyro;
+        turnController = getController();
+        turnController.enableContinuousInput(-180, 180);
+        turnController.setIntegratorRange(-1, 1);
+        turnController.setTolerance(constants.turnPIDToleranceDeg);
+
+        /* MOTOR SETUP */
+        // use arrays to contain each follower motor
+        leftFollowers = new WPI_TalonFX[leftMotorIDs.length-1];
+        rightFollowers = new WPI_TalonFX[rightMotorIDs.length-1];
+        // put each motor in an array corresponding to its type (leader or follower)
+        for(int i = 0; i < leftMotorIDs.length; i++) {
+            // if it's the first one in the array, it's a leader
+            if(i == 0) {
+                // make a new motor with the given ID and put it in the array
+                this.leftLeader = new WPI_TalonFX(leftMotorIDs[i]);
+            }
+            // otherwise it's a follower
+            else {
+                leftFollowers[i-1] = new WPI_TalonFX(leftMotorIDs[i]);
+            }
+        }
+        // do the same for the right motors
+        for(int i = 0; i < rightMotorIDs.length; i++) {
+            if(i == 0) {
+                this.rightLeader = new WPI_TalonFX(rightMotorIDs[i]);
+            }
+            else {
+                rightFollowers[i-1] = new WPI_TalonFX(rightMotorIDs[i]);
+            }
+        }
+        // make speedcontroller groups with the leader and follower motors we got earlier
+        // if there are no followers, do not include the followers in the speedcontrollergroups.
+        if(leftFollowers.length != 0) {
+            leftGroup = new SpeedControllerGroup(leftLeader, leftFollowers);
+        }
+        else {
+            leftGroup = new SpeedControllerGroup(leftLeader);
+        }
+
+        if(rightFollowers.length != 0) {
+            rightGroup = new SpeedControllerGroup(rightLeader, rightFollowers);
+        }
+        else {
+            rightGroup = new SpeedControllerGroup(rightLeader);
+        }
+
         differentialDrive = new DifferentialDrive(leftGroup, rightGroup);
         configureMotors();
     }
     
     // create a required method for subclasses
     public abstract void configureMotors();
 
+    /* COMMANDS */
```

Comment:
why are these multi-line comments?? also i feel like commands is misleading considering we actually use a different thing called commands, so something  like "control methods" might be more self-explanatory.

Reply:
They're multi-line to differentiate them from the other documentation comments (and cuz it's pretty and symmetrical lol)